src/v/cloud_storage/segment_chunk_api.cc

```diff
@@ -304,10 +304,12 @@ ss::future<> chunk_eviction_strategy::close_files(
     }
 
     auto close_results = co_await ss::when_all(fs.begin(), fs.end());
-    for (const auto& result : close_results) {
```

Comment:
nit: unrelated to this PR, but if this function throws, chunk trimming will be stuck as the timer is never armed. Looking at the code, it should not throw but might nice to mark it `noexcept`.